agent/util/subtree_unix.go

```diff
@@ -140,7 +140,19 @@ func parsePs(psOutput string) []int {
 	lines := strings.Split(psOutput, "\n")
 	pids := make([]int, 0, len(lines))
 	for _, line := range lines {
-		pid, err := strconv.Atoi(strings.TrimSpace(line))
+		splitLine := strings.Fields(line)
+		if len(splitLine) < 2 {
+			continue
+		}
+		// Zombie processes are dead processes that haven't been reaped by their parent. They do not consume
+		// resources and will go away by themselves when their parent dies.
+		// This is expected for processes that run in the background, such as a subprocess.exec with Background true.
```

Comment:
So just to make sure (the comment says this but want to validate it's true). The processes that we clean up will kill the zombie processes? Does/can a test cover this or is it more niche behavior?

Reply:
Well, the cleanup could result in the process we've killed being a zombie depending on how it was started and if we've also killed the parent (we won't have killed the parent if the parent is the agent itself). Since the dead child isn't using any resources and it'll disappear on its own when its parent dies, we've done all we need to for it.

This is covered by the `TestKillSpawnedProcs/KillsTrackedProcess` test, which is incidentally what led me to discover the issue 😄 .

Reply:
I've added an explicit test that covers this case.